converter/internal/promtailconvert/validate.go

```diff
@@ -0,0 +1,72 @@
+package promtailconvert
+
+import (
+	"github.com/grafana/agent/converter/diag"
+	promtailcfg "github.com/grafana/loki/clients/pkg/promtail/config"
+	"github.com/grafana/loki/clients/pkg/promtail/targets/file"
+)
+
+// validateTopLevelConfig validates the top-level config for any unsupported features. There may still be some
+// other unsupported features in scope of each config block, which are raised by their respective conversion code.
+func validateTopLevelConfig(cfg *promtailcfg.Config, diags *diag.Diagnostics) {
+	// We currently do not support the new global file watch config. It's an error, since setting it indicates
+	// some advanced tuning which the user likely needs.
+	if cfg.Global.FileWatch != file.DefaultWatchConig {
+		diags.Add(diag.SeverityLevelError, "global/file_watch_config is not supported")
+	}
+
+	// The positions global config is not supported in Flow Mode. It's a warning since the Flow Mode likely
+	// provides a good enough alternative through components' positions files in its own data directory.
+	if cfg.PositionsConfig != defaultPositionsConfig() {
+		diags.Add(
+			diag.SeverityLevelWarn,
```

Comment:
With `SeverityLevelWarn`, we are able to run the agent without bypassing errors in these cases.  I want to be very careful that we are sure we want to be able to do that if the behavior doesn't exactly match promtails for each of these. (same question on other warnings in this validation)

Reply:
That's a good point. I have made this an error and also reviewed the severity level of other diags.